sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySourceBase.java

```diff
@@ -157,9 +157,6 @@ public List<BoundedSource<T>> split(long desiredBundleSizeBytes, PipelineOptions
       LOG.info("Extract job produced {} files", res.extractedFiles.size());
       if (res.extractedFiles.size() > 0) {
         BigQueryOptions bqOptions = options.as(BigQueryOptions.class);
-        // emit this table ID as a lineage source
-        Lineage.getSources()
-            .add(BigQueryHelpers.dataCatalogName(getTableToExtract(bqOptions), bqOptions));
```

Comment:
We can keep Lineage support and instead store the output of `getTableToExtract` in a private variable.

We can store the value when we perform `extractFiles`: https://github.com/apache/beam/blob/3d69dc97c41000d59d43a303912ac12d5f58dce3/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQuerySourceBase.java#L112-L114

Then here, if stored table reference is not null we can add it to Lineage